lib/llm/src/http/service/dynamic_endpoint.rs

```diff
@@ -0,0 +1,112 @@
+// SPDX-FileCopyrightText: Copyright (c) 2024-2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{RouteDoc, service_v2};
+use crate::types::Annotated;
+use axum::{
+    Json, Router,
+    http::{Method, StatusCode},
+    response::IntoResponse,
+    routing::post,
+};
+use dynamo_runtime::instances::list_all_instances;
+use dynamo_runtime::{DistributedRuntime, Runtime, component::Client};
+use dynamo_runtime::{pipeline::PushRouter, stream::StreamExt};
+use std::sync::Arc;
+
+pub fn dynamic_endpoint_router(
+    state: Arc<service_v2::State>,
+    path: Option<String>,
+) -> (Vec<RouteDoc>, Router) {
+    let wildcard_path = "/{*path}";
+    let path = path.unwrap_or_else(|| wildcard_path.to_string());
+
+    let docs: Vec<RouteDoc> = vec![RouteDoc::new(Method::POST, &path)];
+
+    let router = Router::new()
+        .route(&path, post(dynamic_endpoint_handler))
+        .with_state(state);
+
+    (docs, router)
+}
+
+async fn inner_dynamic_endpoint_handler(
+    state: Arc<service_v2::State>,
+    path: String,
+) -> Result<impl IntoResponse, &'static str> {
+    let etcd_client = state.etcd_client().ok_or("Failed to get etcd client")?;
+
+    let instances = list_all_instances(etcd_client)
+        .await
+        .map_err(|_| "Failed to get instances")?;
+
+    let dynamic_endpoints = instances
+        .iter()
+        .filter_map(|instance| instance.http_endpoint_path.clone())
+        .collect::<Vec<String>>();
+
+    let fmt_path = format!("/{}", &path);
+    if !dynamic_endpoints.contains(&fmt_path) {
+        return Err("Dynamic endpoint not found");
+    }
+
+    let rt = Runtime::from_current().map_err(|_| "Failed to get runtime")?;
+    let drt = DistributedRuntime::from_settings(rt)
```

Comment:
I don't think we want to create a new DistributedRuntime for every request. It's intended as a singleton.

I think we have to add it to `State`.

Reply:
+1 for adding it to state, Would make it easy to add the debug config endpoint.